package/src/components/AttachmentPicker/AttachmentPicker.tsx

```diff
@@ -243,10 +243,9 @@ export const AttachmentPicker = React.forwardRef(
       maxNumberOfFiles,
       numberOfAttachmentPickerImageColumns,
       numberOfUploads: selectedFiles.length + selectedImages.length,
-      // `id` is available for Expo MediaLibrary while Cameraroll doesn't share id therefore we use `uri`
       selected:
-        selectedImages.some((image) => image.uri === asset.uri) ||
-        selectedFiles.some((file) => file.uri === asset.uri),
+        selectedImages.some((image) => (image?.uri ? image.uri === asset.uri : false)) ||
```

Comment:
This is significantly improved in https://github.com/GetStream/stream-chat-react-native/compare/attachment-manager-integration...attachment-picker-context-optimizations